src/operator/src/request.rs

```diff
@@ -0,0 +1,233 @@
+            requests,
+            Some(build_db_string(&request.catalog_name, &request.schema_name)),
+            &ctx,
+        )
+        .await
+    }
+
+    /// Handle the request to flush the region.
+    pub async fn handle_region_flush(
+        &self,
+        region_id: RegionId,
+        ctx: QueryContextRef,
+    ) -> Result<AffectedRows> {
+        let request = RegionRequestBody::Flush(FlushRequest {
+            region_id: region_id.into(),
+        });
+
+        self.do_request(vec![request], None, &ctx).await
+    }
+
+    /// Handle the request to compact the region.
+    pub async fn handle_region_compaction(
+        &self,
+        region_id: RegionId,
+        ctx: QueryContextRef,
+    ) -> Result<AffectedRows> {
+        let request = RegionRequestBody::Compact(CompactRequest {
+            region_id: region_id.into(),
+        });
+
+        self.do_request(vec![request], None, &ctx).await
+    }
+}
+
+impl Requester {
+    async fn do_request(
+        &self,
+        requests: Vec<RegionRequestBody>,
+        db_string: Option<String>,
+        ctx: &QueryContextRef,
+    ) -> Result<AffectedRows> {
+        let request_factory = RegionRequestFactory::new(RegionRequestHeader {
+            tracing_context: TracingContext::from_current_span().to_w3c(),
+            dbname: db_string.unwrap_or_else(|| ctx.get_db_string()),
+        });
+
+        let tasks = self
+            .group_requests_by_peer(requests)
+            .await?
+            .into_iter()
+            .map(|(peer, body)| {
+                let request = request_factory.build_request(body);
+                let datanode_manager = self.datanode_manager.clone();
+                common_runtime::spawn_write(async move {
+                    datanode_manager
+                        .datanode(&peer)
+                        .await
+                        .handle(request)
+                        .await
+                        .context(RequestInsertsSnafu)
+                })
+            });
+        let results = future::try_join_all(tasks).await.context(JoinTaskSnafu)?;
+
+        let affected_rows = results.into_iter().sum::<Result<AffectedRows>>()?;
+
+        Ok(affected_rows)
+    }
+
+    async fn group_requests_by_peer(
+        &self,
+        requests: Vec<RegionRequestBody>,
+    ) -> Result<HashMap<Peer, RegionRequestBody>> {
+        let mut inserts: HashMap<Peer, RegionRequestBody> = HashMap::new();
+
+        for req in requests {
+            let region_id = match &req {
+                RegionRequestBody::Flush(req) => req.region_id,
+                RegionRequestBody::Compact(req) => req.region_id,
+                _ => todo!(),
+            };
+
+            let peer = self
+                .partition_manager
+                .find_region_leader(region_id.into())
+                .await
+                .context(FindRegionLeaderSnafu)?;
+
+            inserts.insert(peer, req);
+        }
+
+        Ok(inserts)
+    }
```

Comment:
`flush` and `compact` are not grouped here, `Flush` and `Compact` in `RegionRequestBody` do not support batch

Reply:
Yes, it doesn't do any batch here. The function name is not correct.  Maybe `route_request_by_peer` is better.

Reply:
Fixed in https://github.com/GreptimeTeam/greptimedb/pull/3363/commits/3121fa26bf60ba8709b3a1e2bb07559048b2a5db